osgi.specs/docbook/707/util.converter.xml

```diff
@@ -532,6 +532,27 @@ List&lt;Long&gt; list = c.convert("123").to(new TypeReference&lt;List&lt;Long&gt
         <para>Conversion to <code>Map.Entry</code> from a scalar is not
         supported.</para>
       </section>
+      <section xml:id="util.converter-special.cases.optional">
+        <title>Optional</title>
+
+        <para>Conversions to an Optional is supported as follows.</para>
+        <para><code>null</code> is converted into an empty Optional <code>Optional.empty()</code>.
```

Comment:
This should be an ordered list describing the steps that we go through:

1. Identify the target type of the optional - for `OptionalInt`, `OptionalLong` and `OptionalDouble` this is the wrapper type for the target primitive. For `Optional` it is the generic type.
2. Attempt to convert the value to the target type, throwing a ConversionException if conversion fails
3. If the converted result is null, return the relevant empty optional type.
4. If the converted result is not null, wrap it in the relevant optional value